src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -1038,6 +1086,34 @@ public bool TryGenerateConstant(string possiblyQualifiedName, out IReadOnlyList<
         return false;
     }
 
+    /// <summary>
+    /// Generate code for the named macro, if it is recognized.
+    /// </summary>
+    /// <param name="macroName">The name of the macro. Never qualified with a namespace.</param>
+    /// <param name="preciseApi">Receives the canonical API names that <paramref name="macroName"/> matched on.</param>
+    /// <returns><see langword="true"/> if a match was found and the macro generated; otherwise <see langword="false"/>.</returns>
+    public bool TryGenerateMacro(string macroName, out IReadOnlyList<string> preciseApi)
+    {
+        if (macroName is null)
+        {
+            throw new ArgumentNullException(nameof(macroName));
+        }
```

Comment:
If we're targeting .NET 6+, this could be changed to [`ArgumentNullException.ThrowIfNull`](https://docs.microsoft.com/dotnet/api/system.argumentnullexception.throwifnull)

Reply:
I didn't know about that. Cool. But this generator targets netstandard2.0 so the source generator can run in C# compiler contexts that are hosted by .NET Framework.

Reply:
Maybe we can write our own helper methods?

Reply:
Ah right, no worries then.

In the designer repo we added some throw helpers with #pragmas, something _could_ be useful here, e.g.:
```cs
    /// <summary>
    ///  If <paramref name="argument"/> is <see langword="null"/>, throw an <see cref="ArgumentNullException"/>.
    ///  Otherwise, return the value of <paramref name="argument"/>.
    /// </summary>
    /// <exception cref="ArgumentNullException"/>
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public static T OrThrowIfArgumentIsNull<T>(
        [NotNull] this T? argument,
        [CallerArgumentExpression("argument")] string? paramName = null)
    {
#if NET6_0_OR_GREATER
        ArgumentNullException.ThrowIfNull(argument, paramName);
#else
        ThrowIfArgumentIsNull(argument, paramName);
#endif

        return argument;
    }

    /// <summary>
    ///  If <paramref name="argument"/> is <see langword="null"/>, throw an <see cref="ArgumentNullException"/>.
    /// </summary>
    /// <exception cref="ArgumentNullException"/>
#if NET6_0_OR_GREATER
    [Obsolete("Use ArgumentNullException.ThrowIfNull(...) instead.")]
#endif
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public static void ThrowIfArgumentIsNull(
        [NotNull] object? argument,
        [CallerArgumentExpression("argument")] string? paramName = null)
    {
#if NET6_0_OR_GREATER
        ArgumentNullException.ThrowIfNull(argument, paramName);
#else
        if (argument is null)
        {
            ThrowHelper.ThrowArgumentNullException(paramName);
        }
#endif
```

Reply:
We already have a bunch of helper methods in a `Require` class (in the Microsoft.VisualStudio.Validation package) that we use for this purpose, but as this a source generator and dependencies have a tendency to create new issues, I have to do without them.
While we could re-introduce the helper methods as source code within the project itself, I don't think it's worth it. We hardly have null checks anyway because we use nullable ref annotations everywhere anyway. I only included it here because it was in a public method.
IMO if it took until .NET 6 to add a helper method to the runtime, we can get by with a manual `if` check in a few places. :)

Reply:
You should check out the other performance enhancements in .NET7, its a chapter book! https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7